src/sqlfluff/core/plugin/host.py

```diff
@@ -42,7 +60,24 @@ def get_plugin_manager() -> pluggy.PluginManager:
     # points, this function gets called again - and we only
     # want to load the entry points once!
     _plugin_manager.set(plugin_manager)
-    plugin_manager.load_setuptools_entrypoints(project_name)
+
+    # Discover available plugins an load them individually.
+    # If any fail, log the issue and carry on.
+    for plugin_name, plugin_version in _discover_plugins():
+        plugin_logger.info(f"Loading plugin {plugin_name} version {plugin_version}.")
+        ep = importlib.metadata.entry_points(group=project_name, name=plugin_name)
+        assert ep
```

Comment:
Is the assertion necessary? Do we think it may fail? I wonder if we'd want to catch and report this as an error, too.